Source/WebCore/mathml/MathMLSelectElement.cpp

```diff
@@ -187,10 +187,7 @@ Element* MathMLSelectElement::getSelectedSemanticsChild()
             // If the <annotation-xml> element has an src attribute then it is a reference to arbitrary binary data and it is not clear whether we can display it. Hence we just ignore the annotation.
             if (child->hasAttributeWithoutSynchronization(MathMLNames::srcAttr))
                 continue;
-            // If the <annotation-xml> element has an encoding attribute describing presentation MathML, SVG or HTML we assume the content can be displayed and we stop here.
-            auto& value = child->attributeWithoutSynchronization(MathMLNames::encodingAttr);
-            if (isMathMLEncoding(value) || isSVGEncoding(value) || isHTMLEncoding(value))
-                return child;
```

Comment:
This is to implement the legacy algorithm described on https://developer.mozilla.org/en-US/docs/Web/MathML/Element/semantics ; ideally we should remove this when the MathMLCore flag is on (or if Apple is OK with dropping that implementation) but let's keep this for now.

Reply:
The value of this attribute doesn't strike me as something that should matter for rendering purposes (although unfortunately it does mean something for the HTML parser), but we can keep it around for now, sure.

Reply:
I mean, let's keep this in the present PR, and see if we can remove the implementation mentioned in the note of https://developer.mozilla.org/en-US/docs/Web/MathML/Element/semantics in follow-up PRs.

Reply:
I'm sorry, I'm still a little lost.

This PR is no longer making those changes to `MathMLSelectElement`. Do you want more changes taken out?

It's not clear to me what MathMLCore is or how an MDN reference is supposed to help me. I scanned through https://w3c.github.io/mathml/ and could not find any relevant requirements surrounding `encoding`, other than a bunch of stuff for people writing MathML (and some contradictions such as it apparently containing a MIME type but then not really as `MathML-Content` is not a MIME type).

Reply:
TL;DR: The PR looks good currently!

MathML Core is the spec that is intended to be implemented in browsers: https://w3c.github.io/mathml-core ; the implementation of `<semantics>` there is just "render the first child":

```
semantics > :not(:first-child) {
  display: none;
}
```

The MathML spec at https://w3c.github.io/mathml is not always great for implementation purpose (lacks a lot of details, tests...) but some of the features mentioned there are still implemented in WebKit/Firefox. In particular WebKit still implements the case "when a presentation MathML annotation is present in a semantics element, it may be used as the default rendering of the semantics element, instead of the default rendering of the first child" (https://w3c.github.io/mathml/#mixing_alternate_representations) and more generally the behavior described in the MDN note (but not in any spec) "Legacy MathML specifications allowed renderers to decide the default rendering according to available annotations" (https://developer.mozilla.org/en-US/docs/Web/MathML/Element/semantics).

I believe ideally we just want to simplify the implementation and align it with what is described MathML Core (in particular ignore `encoding` for rendering purpose). But in the past some people at Apple were against unshipping features, that's why I suggested you not to do it in that PR.